pkg/application/app_test.go

```diff
@@ -419,10 +421,59 @@ func Test_genesisExists(t *testing.T) {
 	assert.NoError(err)
 }
 
+func Test_LoadNodeConfig(t *testing.T) {
+	assert := assert.New(t)
+	ap := newTestApp(t)
+
+	err := useViper("node-config-test")
+	assert.NoError(err)
+
+	config, err := ap.LoadNodeConfig()
+	assert.NoError(err)
+	fmt.Println("Config:", config)
+	testVal := viper.GetString("var")
+	fmt.Println("Test val", testVal)
+	assert.Equal("val", testVal)
+}
+
+func Test_LoadNodeConfig_EmptyConfig(t *testing.T) {
+	assert := assert.New(t)
+	ap := newTestApp(t)
+
+	err := useViper("empty-config")
+	assert.NoError(err)
+
+	config, err := ap.LoadNodeConfig()
+	assert.NoError(err)
+	assert.Empty(config)
+}
+
+func Test_LoadNodeConfig_NoConfig(t *testing.T) {
+	assert := assert.New(t)
+	ap := newTestApp(t)
+
+	err := useViper("")
+	// we want to make sure this errors and no config file is read
+	assert.Error(err)
+
+	config, err := ap.LoadNodeConfig()
+	assert.NoError(err)
+	assert.Empty(config)
+}
+
 func newTestApp(t *testing.T) *Avalanche {
 	tempDir := t.TempDir()
 	return &Avalanche{
 		baseDir: tempDir,
 		Log:     logging.NoLog{},
 	}
 }
+
+func useViper(configName string) error {
+	viper.Reset()
+	viper.SetConfigName(configName)
+	viper.SetConfigType("json")
+	viper.AddConfigPath("./../../tests/assets/")
```

Comment:
this config path should be constructed with path join

Reply:
anyway, I think you can generate the files on the fly

Reply:
I think we should keep these as files. I'm not sure how you're suggesting I use path join. Can you let me know what you think the line should be?

Reply:
I was suggesting to avoid slash in paths, filepath.join("..", "..", "tests", "assets")